opentelemetry/src/sdk/trace/span_limit.rs

```diff
@@ -0,0 +1,47 @@
+/// # Span limit
+/// Erroneous code can add unintended attributes, events, and links to a span. If these collections
+/// are unbounded, they can quickly exhaust available memory, resulting in crashes that are
+/// difficult to recover from safely.
+///
+/// To protected against those errors. Users can use span limit to configure
+///  - Maximum allowed span attribute count
+///  - Maximum allowed span event count
+///  - Maximum allowed span link count
+///  - Maximum allowed attribute per span event count
+///  - Maximum allowed attribute per span link count
+///
+/// If the limit has been breached. The attributes, events or links will be dropped based on their
+/// index in the collection. The one added to collections later will be dropped first.
+
+pub(crate) const DEFAULT_MAX_EVENT_PER_SPAN: u32 = 128;
+pub(crate) const DEFAULT_MAX_ATTRIBUTES_PER_SPAN: u32 = 128;
+pub(crate) const DEFAULT_MAX_LINKS_PER_SPAN: u32 = 128;
+pub(crate) const DEFAULT_MAX_ATTRIBUTES_PER_EVENT: u32 = 128;
+pub(crate) const DEFAULT_MAX_ATTRIBUTES_PER_LINK: u32 = 128;
+
+/// Span limit configuration to keep attributes, events and links to a span in a reasonable number.
+#[derive(Copy, Clone, Debug)]
+pub struct SpanLimit {
```

Comment:
I think the spec wants this pluralized

```suggestion
pub struct SpanLimits {
```